### PR TITLE
Email reminder for pending orders

### PR DESCRIPTION
**Pull request by leonardohof**
Resolve opencollective/opencollective#1508
Require opencollective/opencollective-api#2898

### CODE REVIEW

**Comment by leonardohof** [comment]
@znarf Could you review, please?

**Comment by leonardohof** [comment]
image
image
image

File: pages/markOrderAsPaid.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  static SUCCESS = 2;
+  static ERROR = 3;
+
+  async triggerRequest() {
+    try {
+      this.setState({ isRequestSent: true });
+      const res = await this.props.markOrderAsPaid(this.props.id);
+      const orderConfirmed = res.data.markOrderAsPaid;
+      if (orderConfirmed.stripeError) {
+        this.handleStripeError(orderConfirmed);
```
Comment (Betree) [inline pages/markOrderAsPaid.js]: I don't understand how Stripe is related here

File: pages/markOrderAsPaid.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    if (response.paymentIntent) {
+      const stripe = await getStripe(null, account);
+      const result = await stripe.handleCardAction(response.paymentIntent.client_secret);
+      if (result.error) {
+        this.setState({ status: MarkOrderAsPaidPage.ERROR, error: result.error.message });
+      }
+      if (result.paymentIntent && result.paymentIntent.status === 'requires_confirmation') {
+        this.triggerRequest({ id });
+      }
+    }
```
Comment (Betree) [inline pages/markOrderAsPaid.js]: I don't understand how Stripe is related here

File: pages/markOrderAsPaid.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+  markOrderAsPaid = async order => {
+    this.setState({ status: MarkOrderAsPaidPage.SUBMITTING, error: null });
+
+    try {
+      const res = await this.props.markOrderAsPaid(order);
+      const orderConfirmed = res.data.markOrderAsPaid;
+      if (orderConfirmed.stripeError) {
+        this.handleStripeError(orderConfirmed);
+      } else {
```
Comment (Betree) [inline pages/markOrderAsPaid.js]: I don't understand how Stripe is related here

File: server/pages.js [hunk 4]
```diff
@@ -54,6 +54,7 @@ const pages = routes()
   .add('orders', '/:parentCollectiveSlug?/:collectiveType(events)?/:collectiveSlug/orders')
   .add('order', '/:parentCollectiveSlug?/:collectiveType(events)?/:collectiveSlug/orders/:OrderId([0-9]+)')
   .add('confirmOrder', '/orders/:id([0-9]+)/confirm')
+  .add('markOrderAsPaid', '/orders/:id([0-9]+)/markAsPaid')
```
Comment (Betree) [inline server/pages.js]: I think we prefer to enforce snake case for new URLs, @znarf?

```suggestion
  .add('markOrderAsPaid', '/orders/:id([0-9]+)/mark-as-maid')
```

Comment (znarf) [reply server/pages.js]: yes

File: pages/markOrderAsPaid.js (hunk 1 above)
Comment (znarf) [reply pages/markOrderAsPaid.js]: Yes, "Mark as paid" is just for host admins to validate the transaction.

There is absolutely no Stripe Error to expect.

Comment (leonardohof) [reply pages/markOrderAsPaid.js]: Sorry, I used the confirmOrder page as template and forget to remove/change some things...

File: pages/markOrderAsPaid.js (hunk 3 above)
Comment (leonardohof) [reply pages/markOrderAsPaid.js]: Removed.

File: pages/markOrderAsPaid.js (hunk 2 above)
Comment (leonardohof) [reply pages/markOrderAsPaid.js]: Removed.

**Comment by leonardohof** [comment]
I made the rebase from Master.

**Pull request merged**